protos/encodings_v2_1.proto

```diff
@@ -379,23 +379,40 @@ message FixedSizeList {
   CompressiveEncoding values = 2;
 }
 
-// Converts a struct data block into either a fixed-width data block or a variable-width data block
+// Packs a struct containing only fixed-width children into a single fixed-width data block
 //
-// This encoding does not actually compress the data, it just converts the block into something workable
-//
-// This is a transparent encoding.
-//
-// TODO: This needs support for variable-width data blocks and more flexible compression
-//
-// The input is a single struct data block
-// The output is a single fixed-width or variable-width data block
+// The children are concatenated row by row and stored as a single fixed-width buffer. This is
+// the legacy packed struct representation and remains available for backwards compatibility.
 message PackedStruct {
-  // The number of bits per value for each field
+  // The number of bits contributed by each child field in the packed row
   repeated uint64 bits_per_value = 1;
-  // The compression used to store the values data block
+  // The compression used to store the packed fixed-width values
   CompressiveEncoding values = 2;
 }
 
+// Variable-width packed struct encoding (2.1 extension)
+//
+// Each child value is compressed independently before being transposed into
+// a row-major layout. This preserves per-field compression boundaries at the
+// cost of disabling mini-block compression. Readers must prefer this field
+// when present and fall back to the legacy encoding otherwise.
```

Comment:
I'm not sure I understand this last sentence.  Readers should just read whatever was written.  Writers should always choose the newer one.  Right?